File: backend/app/api/v1/schemas/environment_operating_hours.py
from typing import Optional, List
from pydantic import BaseModel


class OperatingHoursDay(BaseModel):
    closed: bool = False
    open: Optional[str] = None
    close: Optional[str] = None


class OperatingHoursConfigIn(BaseModel):
    timezone: str
    week: List[OperatingHoursDay]


class OperatingHoursConfigResponse(BaseModel):
    configured: bool
    timezone: Optional[str] = None
    week: Optional[List[OperatingHoursDay]] = None


class EnvironmentUtilization(BaseModel):
    environment_id: int
    environment_name: str
    configured: bool
    timezone: Optional[str] = None
    total_operating_seconds: float
    booked_operating_seconds: float
    utilization_pct: float


class UtilizationOverview(BaseModel):
    rows: List[EnvironmentUtilization]
    unconfigured_count: int

File: frontend/src/types/environmentOperatingHours.ts
export interface OperatingHoursDay {
  closed: boolean;
  open?: string | null;   // "HH:MM"
  close?: string | null;  // "HH:MM"
}

export interface OperatingHoursConfig {
  configured: boolean;
  timezone?: string | null;
  week?: OperatingHoursDay[] | null;
}

export interface OperatingHoursConfigInput {
  timezone: string;
  week: OperatingHoursDay[];
}

export interface EnvironmentUtilization {
  environment_id: number;
  environment_name: string;
  configured: boolean;
  timezone?: string | null;
  total_operating_seconds: number;
  booked_operating_seconds: number;
  utilization_ratio: number; // 0..1
}

export interface UtilizationOverview {
  rows: EnvironmentUtilization[];
  unconfigured_count: number;
}

export interface UtilizationParams {
  date_from: string; // "YYYY-MM-DD"
  date_to: string;   // "YYYY-MM-DD"
}
